libvast/test/view.cpp

```diff
@@ -45,6 +45,46 @@ TEST(container view) {
   CHECK_EQUAL(i - j, xs.size() - 1);
 }
 
+TEST(set view) {
+  auto xs = set{42, true};
+  auto v = make_view(xs);
+  REQUIRE_EQUAL(v->size(), xs.size());
+  auto i = v->begin();
+  // FIXME: we currently use a std::set instead of a steady set, which relies
+  // on a non-deterministic ordering. This unit test may fail on some platforms
+  // until we have switched to steady_set
+  CHECK_EQUAL(*i, v->at(0));
+  CHECK_EQUAL(*i, make_data_view(42));
+  ++i;
+  CHECK_EQUAL(*i, v->at(1));
+  CHECK_EQUAL(*i, make_data_view(true));
+  ++i;
+  CHECK_EQUAL(i, v->end());
+}
+
+TEST(table view) {
+  auto xs = table{{42, true}, {84, false}};
+  auto v = make_view(xs);
+  REQUIRE_EQUAL(v->size(), xs.size());
+  auto i = v->begin();
+  // FIXME: we currently use a std::table instead of a steady table, which relies
+  // on a non-deterministic ordering. This unit test may fail on some platforms
```

Comment:
> non-deterministic ordering

I assume `std::table` should read `std::map`. Then I have still the same question as with the set.

Reply:
eeb6d0d8120e0be5bb29fc17415faab89c6180de fixes the name, but my question still remains.

Reply:
Yes, that was a typo. 😄